### PR TITLE
Task dev 12

### PR DESCRIPTION
**Pull request by AlexeiRomankov**

### CODE REVIEW

File: task_DEV-12/DEV-12/CoordinateChecker.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        /// <summary>
+        /// Check second element in string.
+        /// </summary>
+        /// <param name="line"></param>
+        /// <returns> Bool result. </returns>
+        private bool IsDigitBetweenOneAndEight(string line)
+        {
+            bool valid = false;
+
+            if (Char.IsDigit(line[1]) && int.Parse(line[1].ToString()) > 0 && int.Parse(line[1].ToString()) < 9)
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/CoordinateChecker.cs]: simplify as return (Char.IsDigit(line[1]) && int.Parse(line[1].ToString()) > 0 && int.Parse(line[1].ToString()) < 9)

File: task_DEV-12/DEV-12/CoordinateChecker.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+            return valid;
+        }
+
+        /// <summary>
+        /// Fill list by letters fields.
+        /// </summary>
+        /// <param name="fields"></param>
+        /// <returns> Filled list. </returns>
+        private List<char> FillList(List<char> fields)
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/CoordinateChecker.cs]: not need to return if fields with be class member

File: task_DEV-12/DEV-12/InputerColor.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            Console.WriteLine("Input color: ");
+
+            while (count != 0)
+            {
+                try
+                {
+                    string mayBeColor = Console.ReadLine();
+
+                    if (!line.Check(mayBeColor))
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/InputerColor.cs]: why throw if we will catch here and reread. we can simply continue

File: task_DEV-12/DEV-12/Program.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                }
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine(messageForNullOrEmtyData);
+            }
+            catch (ArithmeticException)
+            {
+                Console.WriteLine(messageForImpossibleMove);
+            }
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/Program.cs]: What about other exceptions?

File: task_DEV-12/DEV-12/CoordinateChecker.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            for (char i = 'a'; i <= 'h'; i++)
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/CoordinateChecker.cs]: use other lists initializtion, like fileds = new List<char>(){'a','b',....'h'}

File: task_DEV-12/DEV-12/InputerColor.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        /// Input color.
+        /// And five attempts for correct input.
+        /// </summary>
+        /// <param name="line"></param>
+        /// <returns> String of coordination. </returns>
+        public string Input(Checker line)
+        {
+            string messageForIncorrectColor = "Incorrect color. Input \"white\" or \"black\"";
+
+            int count = 5;
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/InputerColor.cs]: has to be constant

File: task_DEV-12/DEV-12/InputerCoordination.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace task_DEV_12
+{
+    class InputerCoordination : IInputer
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/InputerCoordination.cs]: similar to InputerColor, split method into common and type specific parts

File: task_DEV-12/DEV-12/InputerCoordination.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+            string coordinate = string.Empty;
+            int count = 5;
+
+            while (count != 0)
+            {
+                try
+                {
+                    string mayBeCoordinate = Console.ReadLine();
+
+                    if (!line.Check(mayBeCoordinate))
```
Comment (IgorKheidorov) [inline task_DEV-12/DEV-12/InputerCoordination.cs]: no need to throw

File: task_DEV-12/DEV-12/InputerColor.cs (hunk 6 above)
Comment (AlexeiRomankov) [inline task_DEV-12/DEV-12/InputerColor.cs]: it changes on the 33 line

**Pull request merged**